### DISCUSSION TITLE
Unexpected major update

### DISCUSSION THREAD

**Discussion by gajus** [changesets/changesets#1352] [General]
Anyone can explain why changeset is promoting major version https://github.com/gajus/slonik/pull/607 ?

**Comment by Andarist** [comment]
My guess is that `@slonik/dataloaders` is used as a peer dep somewhere but I didn't actually look through all of your packages to confirm that. I'd suggest adding this to your `.changeset/config.json`:
```json
  "___experimentalUnsafeOptions_WILL_CHANGE_IN_PATCH": {
    "onlyUpdatePeerDependentsWhenOutOfRange": true
  }
```

(no, it won't change in patch)

**Reply by gajus** [reply]
For what it is worth, `@slonik/dataloaders` is not referenced anywhere in the codebase. It is a new package.

**Reply by Andarist** [reply]
it could be a transitive problem somehow. It would be cool if you could confirm with that setting - if the problem persists, I could take a deeper look into this.

**Reply by gajus** [reply]
Just added it https://github.com/gajus/slonik/commit/1987a461a80a6f9963377c5047d6c4a07923c1dc

**Reply by gajus** [reply]
Indeed. This created a PR for the correct version update

https://github.com/gajus/slonik/pull/608/files

**Reply by Andarist** [reply]
yeeah, i guess we really need smth like `changeset why` :v

**Comment by multivoltage** [comment]
@Andarist 
Since there a lot of people interested in this behavior, maybe you ca consider to use as default? Honestly I would like to open a PR to add some kind of flag or "fix" this begavior. But the code is to much complex for me.
But I use `changeset` in different project lime other guys.
